apps/api/v2/src/modules/oauth-clients/services/oauth-clients-users.service.ts

```diff
@@ -105,6 +113,14 @@ export class OAuthClientUsersService {
     return this.userRepository.update(userId, body);
   }
 
+  async addUserToTeamEvents(userId: number, organizationId: number) {
+    const orgTeams = await this.organizationsTeamsRepository.findOrgTeams(organizationId);
+
+    for (const team of orgTeams) {
+      await updateNewTeamMemberEventTypes(userId, team.id);
+    }
+  }
```

Comment:
only issue I see is that we have no way of knowing which team is assigned to which oauth client, meaning that if you have multiple oauth clients and teams that are not supposed to be shared, you can end up putting wrong users in teams. let's look into this in a follow up pr

Reply:
Well spotted!

Each OAuth client is tied to a specific organization, and we now the OAuth client ID, so I think we can create a new method that searches teams where parent is orgId and that org has some oauthclient with the known ID.

Does that sound good?

Reply:
Well spotted!

1. Since create org team is behing ApiAuthGuard which is accessible using oauth client ID in the header, we need to update that when a team is created that we store oauth client ID in its `platformOAuthClient` property.
2. Add a new organizations teams repository method that searches all teams by parent = orgId and then where some of oauth clients include one using which the managed user is being created.

Reply:
I tried creating this and all was good until realised that each `PlatformOAuthClient` can only link  to 1 `Team` meaning that it can only store org information and not team, so I am thinking that we need to add a new array `teams: Team[]` in db.

@ThyMinimalDev do you agree its better to have 1st field for `organization` like it is now + 2nd for `teams` instead of having `teams: Team[]` storing both org and teams.

Reply:
I agree, better to have 2 different fields

Reply:
In commit [feat: separate platform teams by OAuth client](https://github.com/calcom/cal.com/pull/15961/commits/80937602eb41bf660f21cad0c6dd1123cb238349):
1. Changed schema prisma by allowing to track which OAuth client was used to create organizations team
2. When creating a managed user we know OAuth client id, so we fetch teams of organization created using that  OAuth client id, and then we add the user to that teams team events marked as `assignAllTeamMembers: true`

Tests also have been updated to check whether or not creating managed users in teams that have `assignAllTeamMembers: true` do not update team event-types of teams created by another OAuth client.